### ISSUE TITLE
TODOs in the code base.

### ISSUE THREAD

**Issue by m-murad** [hawkular/hawkular-android-client#205]
While going through the source code, I noticed there are certain `TODOs` marked in the project. 
@pilhuhn @garg-anuj @danielpassos 
Can we discuss these.
![screen shot 2017-04-14 at 11 49 41 am](https://cloud.githubusercontent.com/assets/17262180/25035025/80a7f502-2108-11e7-9d07-216f5d8c3591.png)

I would like to send a PR for this.

**Comment by pg301** [comment]
@free4murad good idea.  We both can work on it, once we discuss these tasks with @danielpassos @garg-anuj @pilhuhn

**Comment by shubhamvashisht** [comment]
@pg301 @free4murad   i would like to help implementing some tasks too.  @danielpassos @garg-anuj @pilhuhn

**Comment by KeenWarrior** [comment]
we can discuss on gitter
